src/MongoDB/BSON/Decimal.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ *  Copyright 2014-2015 MongoDB, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "hphp/runtime/ext/extension.h"
+#include "hphp/runtime/vm/native-data.h"
+
+#include "../../../mongodb.h"
+#include "../../../utils.h"
+
+#include "Decimal.h"
+
+namespace HPHP {
+
+const StaticString s_MongoBsonDecimal_className("MongoDB\\BSON\\Decimal");
```

Comment:
The extended Json that libmongoc produces, does this though:

```
{ "field" : { "$numberDecimal" : "1234.5678" } }
```

Shouldn't that be $numberDecimal128 as well?